### PR TITLE
fix(VInput/VField): append prepend focus color

### LINKED ISSUE #17763: [Feature Request] prepend/append icon focus color
### Environment
**Vuetify Version:** 3.3.7
**Last working version:** 2.7.0
**Vue Version:** 3.3.4
**Browsers:** Safari 16.5.1, Google Chrome, Safari, Mozilla Firefox
**OS:** Mac OS 10.15.7

### Steps to reproduce
The VInput, icons used to reflect the component color in v2. In v3, this does not work any more.

### Expected Behavior
Icons in component color.

### Actual Behavior
Icons stay gray.

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNrlUstOwzAQ/BXjS0Gq4wNISFVBcOMPOBAOId6CwS+tNwFU9d9ZJwWigqpKHLnFM+MZ72bu1vI6parvQC7kksAn1xBc1kGIZa+alIbP4dDGQI0NgFtoAAneSK0sOPMJCtErHw24i1r6/FjLb4LtIBhlA5soy35FYqxqHCD9JtwjSQgHmn0p92ja6CIymbu2hZynVN+gbQIx2QUD6HgD5pvX02VkcNDS/17EUu82pSBjkZZ6UjA+5hZtIpGBurFn1qeIJNYCYSU2YoXRixl3c1bk5XEhk+BdiouiOJ7dgHNR3EZ05mh2UgJGSzaTczm6Kd+k6jnHwAVfF5d6S/BwCzEgBeOUcq7lE1HKC61bE/ga/z7bYxWAdEheX7FMYxfIelAm+qvT6rQ608ZmmsIVZK8eML5mQDap5XwSQ3b1fkDUVjkknI8JW0i55iGXiB/WmvN6bgHybwYEPHSinWvTqXaoH5OV9E0dNrxvbsRL3ll1Wzq0EHefj/zTyINZcbovgZu5HN7JycMFef8BUqmN/A==)

### PR DESCRIPTION
**Pull request by prashantsinghb**
fixes #17763

## Markup:

```vue
<template>
  <v-app>
    <v-container>
      <v-checkbox label="Checkbox" append-inner-icon="mdi-alert"
        append-icon="mdi-alert"
        prepend-inner-icon="mdi-alert"
        prepend-icon="mdi-alert"
        hint="This is hint"
        color="primary"
      ></v-checkbox>
      <v-autocomplete
        v-model="select"
        :items="items"
        variant="underlined"
        dense
        label="Outlined"
        multiple
        append-inner-icon="mdi-alert"
        append-icon="mdi-alert"
        prepend-inner-icon="mdi-alert"
        prepend-icon="mdi-alert"
        hint="This is hint"
        color="primary"
      ></v-autocomplete>
      <v-combobox
        v-model="select"
        :items="items"
        variant="underlined"
        dense
        label="Outlined"
        multiple
        append-inner-icon="mdi-alert"
        append-icon="mdi-alert"
        prepend-inner-icon="mdi-alert"
        prepend-icon="mdi-alert"
        hint="This is hint"
        color="primary"
      ></v-combobox>
      <v-file-input
        variant="underlined"
        accept="image/*"
        label="File input"
        append-inner-icon="mdi-alert"
        append-icon="mdi-alert"
        prepend-inner-icon="mdi-alert"
        prepend-icon="mdi-alert"
        hint="This is hint"
        color="primary"
      ></v-file-input>
      <v-range-slider
        v-model="value"
        strict
        append-inner-icon="mdi-alert"
        append-icon="mdi-alert"
        prepend-inner-icon="mdi-alert"
        prepend-icon="mdi-alert"
        hint="This is hint"
        color="primary"
      ></v-range-slider>
      <v-select
        :items="items"
        v-model="select"
        append-inner-icon="mdi-alert"
        append-icon="mdi-alert"
        prepend-inner-icon="mdi-alert"
        prepend-icon="mdi-alert"
        hint="This is hint"
        color="primary"
        variant="underlined"
        label="Select"
      />
      <v-slider
        v-model="slider1"
        hint="This is hint"
        color="primary"
        label="color"
        append-inner-icon="mdi-alert"
        append-icon="mdi-alert"
        prepend-inner-icon="mdi-alert"
        prepend-icon="mdi-alert"
      ></v-slider>
      <v-switch
        v-model="model"
        hide-details
        inset
        append-inner-icon="mdi-alert"
        append-icon="mdi-alert"
        prepend-inner-icon="mdi-alert"
        prepend-icon="mdi-alert"
        :label="`Switch: ${model.toString()}`"
        hint="This is hint"
        color="primary"
      ></v-switch>
      <v-text-field
        v-model="msg"
        append-inner-icon="mdi-alert"
        append-icon="mdi-alert"
        prepend-inner-icon="mdi-alert"
        prepend-icon="mdi-alert"
        hint="This is hint"
        color="primary"
        variant="underlined"
        label="Text Field"
      />
      <v-textarea
        v-model="msg"
        append-inner-icon="mdi-alert"
        append-icon="mdi-alert"
        prepend-inner-icon="mdi-alert"
        prepend-icon="mdi-alert"
        hint="This is hint"
        color="primary"
        variant="underlined"
        label="Text Area"
      />
    </v-container>
  </v-app>
</template>

<script>
  export default {
    data () {
      return {
        msg: 'Hello World!',
        items: ['Foo', 'Bar', 'Fizz', 'Buzz'],
        value: [20, 40],
        select: null,
        slider1: 0,
        model: true
      }
    },
  }
</script>

```

### CODE REVIEW

File: packages/vuetify/src/components/VInput/InputIcon.tsx [hunk 1]
```diff
@@ -41,6 +42,7 @@ export function useInputIcon<T extends {}, K extends names = Listeners<T>> (prop
         icon={ props[`${name}Icon`] }
         aria-label={ label }
         onClick={ listener }
+        color={ props.color ?? undefined }
```
Comment (yuwu9145) [inline packages/vuetify/src/components/VInput/InputIcon.tsx:45]: `?? undefined` is unnecessary

Comment (prashantsinghb) [reply packages/vuetify/src/components/VInput/InputIcon.tsx]: Removed unnecessary condition.

**Review by yuwu9145 — DISMISSED** [review]

**Comment by johnleider** [comment]
This is the specification defined in <https://m3.material.io/components/text-fields/specs>.

Realistically, this should be configurable through defaults and added to the md2 and md3 blueprints.

**Comment by johnleider** [comment]
Having icons not change color is actually part of the MD2 and MD3 spec, we just never did it properly. With that being said, I do think that we need a way to handle this. In `v-field` we have the concept of color and baseColor but I'm not sure if that would be applicable here. @KaelWD can you chime in?

**Comment by johnleider** [comment]
Here is an example update that had decent results. @KaelWD thoughts?
```patch
diff --git a/packages/vuetify/src/components/VInput/VInput.tsx b/packages/vuetify/src/components/VInput/VInput.tsx
index 205273d38..66714c818 100644
--- a/packages/vuetify/src/components/VInput/VInput.tsx
+++ b/packages/vuetify/src/components/VInput/VInput.tsx
@@ -6,6 +6,7 @@ import { useInputIcon } from '@/components/VInput/InputIcon'
 import { VMessages } from '@/components/VMessages/VMessages'
 
 // Composables
+import { useTextColor } from '@/composables/color'
 import { makeComponentProps } from '@/composables/component'
 import { makeDensityProps, useDensity } from '@/composables/density'
 import { IconValue } from '@/composables/icons'
@@ -41,6 +42,8 @@ export const makeVInputProps = propsFactory({
     type: Boolean,
     default: true,
   },
+  color: String,
+  baseColor: String,
   prependIcon: IconValue,
   hideDetails: [Boolean, String] as PropType<boolean | 'auto'>,
   hint: String,
@@ -105,6 +108,12 @@ export const VInput = genericComponent<VInput
Slots>()({
       validationClasses,
     } = useValidation(props, 'v-input', id)
 
+    const { textColorClasses, textColorStyles } = useTextColor(co
mputed(() => {
+      return props.error || props.disabled ? undefined
+        : props.focused ? props.color
+        : props.baseColor
+    }))
+
     const slotProps = computed<VInputSlot>(() => ({
       id,
       messagesId,
@@ -148,10 +157,14 @@ export const VInput = genericComponent<VInpu
tSlots>()({
             },
             densityClasses.value,
             rtlClasses.value,
+            textColorClasses.value,
             validationClasses.value,
             props.class,
           ]}
-          style={ props.style }
+          style={[
+            textColorStyles.value,
+            props.style,
+          ]}
         >
           { hasPrepend && (
             <div key="prepend" class="v-input__prepend">
```

**Comment by prashantsinghb** [comment]
@johnleider - Have implemented your suggestion with `.v-field__input color: fieldText`, seems to working fine with every input field.

But not sure about VSwitch behavior as visible in the attached video.

https://github.com/vuetifyjs/vuetify/assets/57935926/c0c15f54-0fdc-4fa1-99fb-3fb4ef70b4d5

**Comment by KaelWD** [comment]
That's definitely incorrect, if you want to match v2 it should only apply to components that use VField (so not sliders or selection controls). Should also not apply to hint text. 

> Having icons not change color is actually part of the MD2 and MD3 spec, we just never did it properly. 

Yeah this is MD1 only really, do we actually need it? Not sure how you'd go about switching it on or off.

**Comment by johnleider** [comment]
I think it's a good thing to add. A bug actually made it work when using hex colors a few versions ago.

**Comment by prashantsinghb** [comment]
> This can be a feature but does need to go to the dev branch. 

Will move to dev branch.

> It needs to be an option through a new property or existing if it makes sense.

@johnleider  - I could not find any existing appropriate property for this behavior
Can I add `enableFocusColor` as boolean prop for this functionality? Any suggestion is most welcome.

**Comment by johnleider** [comment]
Please double check the logic that determines the color. It's not checking for the focus state in a few components.

**Comment by johnleider** [comment]
This Pull Request is being closed due to inactivity.

If you have any additional questions, please reach out to us in our [Discord community](https://community.vuetifyjs.com).

**Comment by andreclemente** [comment]
Hi, any news regarding this pull request?

**Pull request closed without merging**